pflake8/__init__.py

```diff
@@ -51,16 +54,42 @@ class DivertingSafeConfigParser(ConfigParserTomlMixin, configparser.SafeConfigPa
     pass
 
 
-class ModifiedConfigFileFinder(flake8.options.config.ConfigFileFinder):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.project_filenames = self.project_filenames + ('pyproject.toml',)
-
-
 configparser.RawConfigParser = DivertingRawConfigParser
 configparser.ConfigParser = DivertingConfigParser
 configparser.SafeConfigParser = DivertingSafeConfigParser
-flake8.options.config.ConfigFileFinder = ModifiedConfigFileFinder
+
+
+class FixFilenames(ast.NodeTransformer):
+    tuple_of_interest = ("setup.cfg", "tox.ini", ".flake8")
+
+    def visit_Tuple(self, node: ast.Tuple) -> Any:
+        if all(isinstance(el, ast.Constant) for el in node.elts) and set(
+            self.tuple_of_interest
+        ) == {el.value for el in node.elts}:
+            node.elts.append(ast.Constant(value="pyproject.toml"))
+            ast.fix_missing_locations(node)
+        return node
+
+    @classmethod
+    def apply(cls, module=flake8.options.config):
+        filename = module.__file__
+
+        exec(
+            compile(
+                cls().visit(
+                    ast.parse(
+                        Path(filename).read_text(encoding='UTF-8'), filename=filename
+                    )
+                ),
+                filename=filename,
+                mode="exec",
+            ),
+            module.__dict__,
+        )
+
+
+FixFilenames.apply()
```

Comment:
It seems like if the patching fails (e.g. if `flake8` devs refactor again) this would silently run `flake8` without `prproject.toml`. Could we instead set some kind of return flag and signal whether we found the `tuple_of_interest` and patched it? This would make it so that `pflake8` would fail fast and users would be immediately alerted the next time flake8 makes a breaking change upstream rather than the command running but with the wrong configuration settings.

Reply:
We should pin version in requirements.

Reply:
Do you mean pin to exact versions of `flake8`? I suppose we could if we released `pyproject-flake8` after every `flake8` release, otherwise we'd be holding users up from new `flake8` versions.

Perhaps with a GitHub Action and some integrations tests, we could set up the repo to confidently automatically publish new versions when `flake8` releases on PyPI.

Reply:
I think detecting failure to patch `tuple_of_interest` is much better than capping/pinning versions. https://iscinumpy.dev/post/bound-version-constraints/#tldr gives a good summary of the problems that capping/pinning Python dependencies causes.

Reply:
I'm quite split on the issue, currently leaning towards maybe releasing a pyproject-flake8 version per flake8 version. Eventually flake8 may always break compatibility in the future...